test-fastapi.sh

```diff
@@ -0,0 +1,24 @@
+#! /usr/bin/env bash
```

Comment:
if you're brave you could probably implement this inside the `Makefile`.

Makefile even has it's [own logic](https://stackoverflow.com/questions/5553352/how-do-i-check-if-file-exists-in-makefile-so-i-can-delete-it) for checking if a file or directory exists.

If it's a pain, leave it for now and I'll fix in future.

Reply:
otherwise maybe move this into `tests` to avoid cluttering up the root directory.

Reply:
make the `fastapi` directory and any other related files should also live in `tests`.

Reply:
About putting it in the `Makefile`, I tried first to put all the bash script in it, but I couldn't.

I tried several things, but I couldn't get the inline variables to work, the: `latest_tag_commit` and `latest_tag`. I also tried getting rid of them and "sub-bashing" the commands, nesting them, but couldn't get that to work either :disappointed: 

If you know a Make trick for that, let me know :sweat_smile: 

About the logic to check for the directory, I first had a:

```Makefile
fastapi:
	git clone https://github.com/tiangolo/fastapi.git

.PHONY: test-fastapi
test-fastapi: fastapi
	bash tests/test-fastapi.sh
```

To use the automatic directory target `fastapi`. But I removed it and put it all in the bash script, trying not to clutter the `Makefile`.

If you prefer it like that, with the `git clone` in the `Makefile`, I can put it there.

The rest of the Bash script... I don't really know what else to try to make those variables to work :thinking:

Reply:
I thought you would probably prefer the extra Make target, so I just went ahead and refactored that.

Reply:
I moved the script to `./tests/`.

But I couldn't move the `fastapi` repo clone to under `tests`, because `tests` is already a package, and it seems the imports get confused when trying to run tests inside of a directory inside of a `tests` directory.

I can also clone it to a `./tmp/fastapi` dir to make it explicit. Or a `./.tmp/fastapi` to make it a hidden dir.